### PR TITLE
Feat/event management

### PR DESCRIPTION
**Pull request by misterM125**
## Overview

This PR aims to implement event management into the app, using the event overview screen to edit an event or change its status

### Features

- **Event editing**: The owner of an event can now edit his event by clicking on the new "edit event" button of the event overview screen, allowing him to modify any information about an event, which will then be updated in the park overview if he validates the changes

- **Event deletion**: Also when clicking the "edit event" button, the owner will be given the option to delete his event, which will delete said event from the database and remove it from the event list in the park

- **Event status**: A new Enum type `EventStatus` has been created, this enum is to be used in the new `status` field of an event to change its current status (`CREATED`, `STARTED` or `ENDED`), every newly created event will start as `CREATED`:
With this new field, an owner will now be offered different options on the event overview screen depending on the status of his event, allowing him to modify it whenever he chooses so. He can: 1. Start his event if the event has just been created or 2. end said event after it has been started, ending an event will essentially have the same effect as deleting it

- Park overview screen: a few changes (mainly ui) have been made to the overview screen to conform to these changes and to make use of the `EventStatus`:
When an event is created, a red icon will be displayed to indicate that the event hasn't started yet, when an event has been started the icon will change and turn green to indicate an ongoing event. 
The status text over the event title has also been refactored to better indicate if an event is starting soon or is already ongoing

### Implementation
- The `AddEventScreen` has been refactored to allow event editing by adding a new boolean parameter, which will make the screen a bit different and change its functionalities, as such the screen is considered as a different screen in the navigation tree (Screen.EditEvent)

- The function `deleteEvent` has been added to the event mvvm to allow for event deletion

- `EventBottomBar` has been refactored for the owner to have an edit button as well as a status button ont the event overview

## How to test
- Go to the park overview screen by clicking on a marker and try creating an event
- You should then see it displayed in the park overview with a red marker, if the event's start date is close (less than an hour) you should see "starting soon!" displayed as a status
- going to the event overview screen of your newly created event, you can choose to click on the "edit event" button which will offer you to change any field of your event or delete your event entirely
- you can also choosse to click on the "start your event" button, which will change its status on the park overview screen
- you can then go back to the park overview screen to end your event by clicking on the "end event" button

### Additonal info

- The `ENDED` state of an event should in theory never be seen or even possible to attain, since the event will be deleted from the database when ended, but it still has been added as an edge case, mainly for events already in the database
- TopBar and BottomBar tests have been refactored to conform to the new edit event screen, it is now mostly ignored in both, as behavior can be a bit unstable
- This is more of a note to anyone who would want use or add tests for the edit event screen: `editEvent` boolean parameter of the add event screen should only be set to true if and only if the the `eventviemodel`'s `currentevent` is not null, else the screen will crash

### CODE REVIEW

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 1]
```diff
@@ -19,9 +19,13 @@ import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.material.icons.Icons
-import androidx.compose.material.icons.filled.AccountCircle
+import androidx.compose.material.icons.automirrored.filled.DirectionsRun
+import androidx.compose.material.icons.filled.DirectionsRun
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:25]: Import: this unused import could be removed

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 2]
```diff
@@ -392,26 +396,40 @@ fun EventItemList(eventViewModel: EventViewModel, navigationActions: NavigationA
  */
 @Composable
 fun EventItem(event: Event, eventViewModel: EventViewModel, navigationActions: NavigationActions) {
+  val (statusColor, statusIcon) =
+      when (event.status) {
+        EventStatus.CREATED -> Pair(Color.Red, Icons.Default.Timelapse)
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:401]: UI: the green and red color are a little too different from the rest of the UI. For the red, I suggest that we use the same color as the one for the picture. (you can see it in PR#179 , the last screenshot)

For the green, I do not think that we already have a instance of green in our UI, but I think that making it a darker green could be better for the UI.

File: app/src/main/java/com/android/streetworkapp/utils/DateUtils.kt [hunk 3]
```diff
@@ -73,13 +78,18 @@ fun dateDifference(endTimestamp: String): String {
 
   val hours = duration.toHours() % 24
 
+  val statusText =
+      when (event.status) {
+        EventStatus.STARTED -> "event has begun!"
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/utils/DateUtils.kt:83]: Typo: to stay coherent, perhaps the first letter should be capital, such as what has been done in line 85

(The same applies to the line 84)

In addition, the String.xml file could be used for such strings.

File: app/src/main/java/com/android/streetworkapp/ui/event/AddEvent.kt [hunk 4]
```diff
@@ -199,26 +211,50 @@ fun AddEventScreen(
...
+                          if (editEvent) {
+                            Text("Edit event")
+                          } else {
+                            Text("Add new event")
+                          }
+                        }
+
+                    if (editEvent) {
+                      Button(
+                          colors = ColorPalette.BUTTON_COLOR.copy(containerColor = Color.Red),
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/event/AddEvent.kt:247]: (Same as last comment about colors)

File: app/src/main/java/com/android/streetworkapp/ui/event/EventOverview.kt [hunk 5]
```diff
@@ -310,6 +313,65 @@ fun LeaveEventButton(
...
+      when (event.status) {
+        EventStatus.CREATED -> Pair(EventStatus.STARTED, "Start Event")
+        EventStatus.STARTED -> Pair(EventStatus.ENDED, "End Event")
+        EventStatus.ENDED -> Pair(EventStatus.ENDED, "Event Ended")
+      }
+
+  val buttonColor =
+      when (event.status) {
+        EventStatus.CREATED -> ColorPalette.BUTTON_COLOR.copy(containerColor = Color.Green)
+        EventStatus.STARTED -> ColorPalette.BUTTON_COLOR.copy(containerColor = Color.Red)
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/event/EventOverview.kt:352]: (Same as last comment about colors)

**Review by alvaro1080 — CHANGES_REQUESTED** [review]
Thank you for this Pull Request! Being able to better manage the event is a really valuable feature.

I have a few concerns about the colors used, and a few details I would like to discuss, but over all it is a great Pull Request!

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt (hunk 2 above)
Comment (alvaro1080) [reply app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:401]: (I just realized: I think that the best possible green is the one that is used in this page. Look att the "Open" logo, just under the title of the PR. I think this is the best for our UI)

**Comment by misterM125** [comment]
I agree with all proposed changes, @alvaro1080 let me know what you think about the new green i used, i tried to get something as possible to what you had in mind

**Review by alvaro1080 — APPROVED** [review]
Thank you for the changes!
